Sources/WalletConnectModal/Modal/RecentWalletStorage.swift

```diff
@@ -9,23 +9,44 @@ final class RecentWalletsStorage {
 
     var recentWallets: [Listing] {
         get {
-            guard
-                let data = defaults.data(forKey: "recentWallets"),
-                let wallets = try? JSONDecoder().decode([Listing].self, from: data)
-            else {
-                return []
-            }
-            
-            return wallets
+            loadRecentWallets()
         }
         set {
-            guard
-                let walletsData = try? JSONEncoder().encode(newValue)
-            else {
-                return
+            saveRecentWallets(newValue)
+        }
+    }
+    
+    func loadRecentWallets() -> [Listing] {
+        guard
+            let data = defaults.data(forKey: "recentWallets"),
+            let wallets = try? JSONDecoder().decode([Listing].self, from: data)
+        else {
+            return []
+        }
+        
+        return wallets.filter { listing in
+            guard let lastTimeUsed = listing.lastTimeUsed else {
+                assertionFailure("Shouldn't happen we stored wallet without `lastTimeUsed`")
```

Comment:
Could you explain why we are storing without lastTimeUsed please?

Reply:
It should never happen that we store it without `lastTimeUsed`, so I've added this just as an extra check If we ever do.